src/metadata.rs

```diff
@@ -33,7 +33,7 @@ macro_rules! box_result {
 /// `fetch_metadata` is the generic, top-level function that is used by the main
 /// function to fetch metadata. The configured provider is passed in and this
 /// function dispatches the call to the correct provider-specific fetch function
-pub fn fetch_metadata(provider: &str) -> errors::Result<Box<providers::MetadataProvider>> {
+pub fn fetch_metadata(provider: &str) -> errors::Result<Box<dyn providers::MetadataProvider>> {
```

Comment:
Why is this fix needed? Is it due to new Rust version?

Reply:
presumably so. For me on "rustc 1.50.0 (cb75ad5db 2021-02-10)", `cargo build --verbose` shows:

```shell
warning: trait objects without an explicit `dyn` are deprecated
  --> src/metadata.rs:36:61
   |
36 | pub fn fetch_metadata(provider: &str) -> errors::Result<Box<providers::MetadataProvider>> {
   |                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn providers::MetadataProvider`
   |
   = note: `#[warn(bare_trait_objects)]` on by default

```

Reply:
Ah, I see :+1: I guess both fixes can be committed together as `src: fix warnings found by "cargo build"` or something :)

Reply:
done